src/Compilers/Core/Portable/DiagnosticAnalyzer/ProgramFilesAnalyzerPathResolver.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis;
+
+/// <summary>
+/// This <see cref="IAnalyzerPathResolver"/> implementation is used to handle analyzers that
+/// exist in global install locations on Windows. These locations do not need to be shadow
+/// copied because they are read-only and are not expected to be updated. Putting this resolver
+/// before shadow copy will let them load in place.
+/// </summary>
+internal sealed class ProgramFilesAnalyzerPathResolver : IAnalyzerPathResolver
+{
+    public string ProgramFilesPath { get; }
+    public string DotNetPath { get; }
+    public string VisualStudioPath { get; }
+
+    public ProgramFilesAnalyzerPathResolver()
+    {
+        ProgramFilesPath = Environment.GetFolderPath(Environment.SpecialFolder.ProgramFiles);
+        DotNetPath = Path.Combine(ProgramFilesPath, "dotnet");
+        VisualStudioPath = Path.Combine(ProgramFilesPath, "Microsoft Visual Studio");
```

Comment:
this feels... unsafe.  like VS can be arbitrarily installed to different paths right?

Reply:
Yes it can be installed to other paths. That isn't unsafe though, it just means they wouldn't get the same optimization. The overall mentality is that Program Files is not a directory that is edittng as part of your inner REPL. I originally considered saying that _anything_ under Program Files wouldn't be shadow copied but decided to be more conservative and scope to a few well known places. 

I also attempted to let the value be thread through from the VS layer and that was quite the mess. The VS layer does have access to this but it is not done in a way that can be thread through easily. 

I'm fine with removing this particular directory from the consideration. We don't get a tone of value from it, the valeu is in the `dotnet` directory.